### PR TITLE
Rtc ds3231

### PR DESCRIPTION
**Pull request by Klaus-Michael**
Add documentation to add a RTC DS3231 device to the i2c Bus and to automatically sync the system time with the rtc time during startup.
This ensures that the timestamps of the Rounds are correct if the System is running without any internet connection.

### CODE REVIEW

File: doc/Software Setup.md [hunk 1]
```diff
@@ -133,6 +133,64 @@ Install the Python library:
...
+```
+echo ds3231 0x68 > /sys/class/i2c-adapter/i2c-1/new_device
+while ! hwclock -s
+do
+   echo "Failed to sync RTC to local time."
+done
+   echo "Successfully synced RTC to local time"
+```
+
+disable to the usage of the fake-hwclock by executing
```
Comment (pulquero) [inline doc/Software Setup.md]: 'disable fake-hwclock by executing'

**Comment by pulquero** [comment]
It looks like this device includes an onboard temperature sensor. In the future, it would be worth seeing if the temperature can be added as another type of sensor that RH can read.

**Comment by HazardCreative** [comment]
I like the reasoning behind this, but some will not want to spend money on yet another component. Can the pi report if it didn't sync? If so and no RTC is present, maybe this could be automated by reading the system time of the first connected client? I think it can be assumed that computers and phones will carry a usable timestamp. Manually setting the system clock is also a possibility.

**Comment by BoLtWD** [comment]
The price is not worth thinking about, as long as it helps

**Comment by BoLtWD** [comment]
One thing I would like to know, if you do fit a Rtc ds3231 as above code then make an image of your card for someone other to use that has no Rtc ds3231 what would be the outcome? would it revert back to the fake clock or would you have to reliable it?

File: doc/Hardware Setup.md [hunk 2]
```diff
@@ -61,8 +61,11 @@ The pins in the green box is what were already used by the timer. The pins in th
 ### BME280 Temperature support (optional)
 Attach to the I2C bus and 5V pins.
 
------------------------------
+### DS3231 I2C Real Time Clock rtc  (optional)
+Attach to the I2C bus and 3.3V pins.
```
Comment (pulquero) [inline doc/Hardware Setup.md]: From what I've read you can use 5v or 3.3v

**Comment by HazardCreative** [comment]
> The price is not worth thinking about, as long as it helps
Maybe for you, but some build these open source systems specifically because they are a "budget" option where professional-grade multi-pilot timing is concerned.

**Comment by pulquero** [comment]
As far as I understand it, this is completely optional and marked in the documentation as such. So, if you don't want to spend the money, just skip those parts of the documentation and run as normal. If you want to spend the money, follow the documentation and run as normal. The only difference will be who/what is responsible for setting the system time.

**Comment by BoLtWD** [comment]
@HazardCreative  Sorry if I have offended, my comment was not meant to come over to upset you. Like @pulquero has commented on I believed this was optional as per comment fc8aaac. 

Based on my comment "The price is not worth thinking about, as long as it helps"
All modules have a price tag but all are depending on personal preference how you want to develop your timer. Most can be added at a later date adding more functionality like building blocks splitting the cost. I think the way things have developed is fantastic giving options for 1-8 Pilots, LEDs, temperature, voltage, Split-timing support and many more all helping race directors create a timer to there needs. 

A concern I did have is if you have something added this could make diagnostics of issues harder as everyone may have a different setups. This why I was asking as above how easy it was to revert back if needed, the ability to turn off modulus or disable to help in diagnostics if needed.

Would installing the Rtc ds3231 and then made a copy of my card for someone with a stranded setup work?  If this would not work then it would be nice to include how you can revert the rack to the fake clock in the documentation, I believe other modulus are auto detection.

Again sorry if my wording has been taken the wrong way.

**Comment by HazardCreative** [comment]
No offense taken and your question is a good one.

**Comment by pulquero** [comment]
I've just installed one of these and found the only setup I needed was to add the following to /boot/config.txt:
dtoverlay=i2c-rtc,ds3231
Note: I run NTP.

**Comment by BoLtWD** [comment]
> I've just installed one of these and found the only setup I needed was to add the following to /boot/config.txt:
> dtoverlay=i2c-rtc,ds3231
> Note: I run NTP.

I take it running it this way is more of a detection then an install, if that is the case then when and if you remove the module it will revert back.

**Comment by BoLtWD** [comment]
> I've just installed one of these and found the only setup I needed was to add the following to /boot/config.txt:
> dtoverlay=i2c-rtc,ds3231
> Note: I run NTP.

Just installed a RTC as quoted text, worked without problem. I have not tried it but wondering what would happen if you have setup the RTC yet disconnected the module?

**Comment by HazardCreative** [comment]
Please do try it and report back.

**Comment by BoLtWD** [comment]
My set up is as pulquero and just disconnected the RTC, I can see no problem on the bench however this is connected to the internet.

**Comment by HazardCreative** [comment]
Excellent, thank you. Please also try a non-connected test when possible.

**Comment by HazardCreative** [comment]
Looks like this PR should be amended with the various comment findings. I'd like to get that done and pulled in.

**Comment by szafranski** [comment]
So it works with the newest release?

**Comment by BoLtWD** [comment]
@szafranski The issue I had #210 has been fixed

**Comment by szafranski** [comment]
Thanks

**Comment by sl33pydog** [comment]
I've followed this procedure using a ds3231 and a cr1220 coin cell on my pcb and it works perfectly.  

https://learn.adafruit.com/adding-a-real-time-clock-to-raspberry-pi/set-rtc-time

Tested it first online and then took it offline, left it for 2 days, started it up, ran a race and saw it followed the time perfectly.  I made the image using the RH OTA, had to fix the LEDs with szafranski's help since I'm using a pi 3B and then followed the adafruit instructions above to fully get the pi to use the RTC instead of the fake time.  Going to be testing more but I think it works ok right now with it.  Also using two ina219 sensors (different addresses) along with a bme280 all work fine.

**Comment by HazardCreative** [comment]
Does the information in https://github.com/RotorHazard/RotorHazard/pull/134#issuecomment-510685496 still need added into this PR?

**Comment by sl33pydog** [comment]
I guess when I installed it my image didn't have ntp on by default.  Could also be that my ds3231 didn't have the time set.  I'll do another fresh install.  Maybe all that would need to be done by the user or added is sudo hwclock -w if you wanted to make sure the module had the right time and date.

**Comment by szafranski** [comment]
If you were using ota - ntp is automatically installed. Just wanna this to
be mentioned in case of some throubleshooting etc.

W dniu wt., 31.03.2020 o 17:30 sl33pydog <notifications@github.com>
napisał(a):

**Pull request closed without merging**